backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -445,6 +482,74 @@ export class FeatureFlagService {
     return includedFeatureFlags;
   }
 
+  public async importFeatureFlags(
+    featureFlagFiles: IFeatureFlagFile[],
+    logger: UpgradeLogger
+  ): Promise<IImportError[]> {
+    logger.info({ message: 'Import feature flags' });
+    const validatedFlags = await this.validateImportFeatureFlags(featureFlagFiles, logger);
+
+    const fileStatusArray = featureFlagFiles.map((file) => {
+      const validation = validatedFlags.find((error) => error.fileName === file.fileName);
+      const isCompatible = validation && validation.compatibilityType !== FF_COMPATIBILITY_TYPE.INCOMPATIBLE;
+
+      return {
+        fileName: file.fileName,
+        error: isCompatible ? null : FF_COMPATIBILITY_TYPE.INCOMPATIBLE,
+      };
+    });
+
+    const validFiles: FeatureFlagValidation[] = fileStatusArray
+      .filter((fileStatus) => fileStatus.error === null)
+      .map((fileStatus) => {
+        const featureFlagFile = featureFlagFiles.find((file) => file.fileName === fileStatus.fileName);
+        return JSON.parse(featureFlagFile.fileContent as string);
+      });
+
+    const createdFlags = await getConnection().transaction(async (transactionalEntityManager) => {
+      return await Promise.all(
+        validFiles.map(async (featureFlag) => {
+          const newFlag = await this.addFeatureFlagInDB(
+            this.featureFlagValidatorToFlag(featureFlag),
+            logger,
+            transactionalEntityManager
+          );
+
+          const featureFlagSegmentInclusionList = featureFlag.featureFlagSegmentInclusion.map(
+            (segmentInclusionList) => {
+              segmentInclusionList.list.id = uuid();
+              segmentInclusionList.flagId = newFlag.id;
+              return segmentInclusionList;
+            }
+          );
+          const inclusionDoc = await this.addList(
```

Comment:
inclusionDoc and exclusionDoc be added in a `Promise.all`

Reply:
code updated